src/cloud-api-adaptor/test/provisioner/provision.go

```diff
@@ -287,46 +287,46 @@ func (lio *KbsInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, pro
 	return nil
 }
 
-func (p *KeyBrokerService) GetKbsPodIP(ctx context.Context, cfg *envconf.Config) (string, error) {
+func (p *KeyBrokerService) GetKbsServiceIP(ctx context.Context, cfg *envconf.Config) (string, error) {
 	client, err := cfg.NewClient()
 	if err != nil {
 		return "", err
 	}
 
 	namespace := "coco-tenant"
-	deploymentName := "kbs"
-
-	err = AllPodsRunning(ctx, cfg, namespace)
-	if err != nil {
-		err = fmt.Errorf("All pods are not running: %w\n", err)
-		log.Errorf("%v", err)
-		return "", err
-	}
+	serviceName := "kbs"
 
 	resources := client.Resources(namespace)
-	podList := &corev1.PodList{}
-	err = resources.List(context.TODO(), podList)
-	if err != nil {
-		err = fmt.Errorf("Error listing pods: %w\n", err)
-		log.Errorf("%v", err)
+
+	services := &corev1.ServiceList{}
+	if err := resources.List(context.TODO(), services); err != nil {
 		return "", err
 	}
+	var clusterIP string = ""
+	var nodePort int32 = 0
+	for _, service := range services.Items {
+		if service.ObjectMeta.Name == serviceName {
+			// Ensure the service is of type NodePort
+			if service.Spec.Type != corev1.ServiceTypeNodePort {
+				return "", fmt.Errorf("Service %s is not of type NodePort", "kbs")
+			}
 
-	var matchingPod *corev1.Pod
-	for i := range podList.Items {
-		pod := &podList.Items[i]
-		if pod.Labels["app"] == deploymentName {
-			matchingPod = pod
-			break
+			// Extract ClusterIP
+			clusterIP = service.Spec.ClusterIP
+			if clusterIP == "" {
+				return "", fmt.Errorf("Service %s does not have a ClusterIP", serviceName)
+			}
+			// Extract NodePort
+			if len(service.Spec.Ports) > 0 {
+				nodePort = service.Spec.Ports[0].NodePort
+			} else {
+				return "", fmt.Errorf("NodePort is not configured for Service %s", "kbs")
+			}
 		}
 	}
 
-	if matchingPod == nil {
-		return "", fmt.Errorf("No pod with label selector found")
-	}
-
-	fmt.Printf("Pod IP: %s\n", matchingPod.Status.PodIP)
-	return matchingPod.Status.PodIP, nil
+	serviceString := fmt.Sprintf("%s:%d", clusterIP, nodePort)
```

Comment:
I think this is still the address in CNI network but not NodeIP:port?
For example, in my libvirt env.
```
# kubectl get svc kbs
NAME   TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
kbs    NodePort   10.96.46.215   <none>        8080:30194/TCP   3d2h

# kcli list vm
+------------------------+--------+-----------------+------------+-----------+---------+
|          Name          | Status |        Ip       |   Source   |    Plan   | Profile |
+------------------------+--------+-----------------+------------+-----------+---------+
|  peer-pods-ctlplane-0  |   up   | 192.168.122.123 | ubuntu2204 | peer-pods |  kvirt  |
|   peer-pods-worker-0   |   up   | 192.168.122.108 | ubuntu2204 | peer-pods |  kvirt  |
```
I think we should have `http://192.168.122.108:30194` but not `http://10.96.46.215:8080`.